### PR TITLE
test: keyboard navigation using tab

### PR DESCRIPTION
**Pull request by LiKang6688**

### CODE REVIEW

File: test/integration/keyboard.integration.ts [hunk 1]
```diff
@@ -17,6 +17,17 @@ test.describe('Tests served by: fixture-file rendering by Nebula', () => {
...
+  const isSVG = async () => {
+    const index = (await page.locator('*:focus').innerHTML()).match(/svg/)?.index;
+    if (index && index > 0) return true;
+    return false;
+  };
+  const isOption = async () => {
+    const index = (await page.locator('*:focus').innerHTML()).match(/option/)?.index;
+    if (index && index > 0) return true;
+    return false;
+  };
```
Comment (haxxmaxx) [inline test/integration/keyboard.integration.ts:30]: There aren't really any util functions defined in this file. Do they fit into some other file or maybe create a new one?

**Review by haxxmaxx — COMMENTED** [review]
I don't mind adding the test. Maybe we need to do some shift+tab as well. But this is deviating from the the test structure in the arrow key test. Would be nice if we can write new tests in a similar way. Doesn't have to be as exactly as the arrow test. But much of the same things are going on: Do something (in this case tab) then check where we are. Should be possible to define the steps and then loop over them

File: test/integration/keyboard.integration.ts (hunk 1 above)
Comment (LiKang6688) [reply test/integration/keyboard.integration.ts:30]: Okay, I will create a new file to relocate them.

**Comment by LiKang6688** [comment]
> I don't mind adding the test. Maybe we need to do some shift+tab as well. But this is deviating from the the test structure in the arrow key test. Would be nice if we can write new tests in a similar way. Doesn't have to be as exactly as the arrow test. But much of the same things are going on: Do something (in this case tab) then check where we are. Should be possible to define the steps and then loop over them

I dislike the loop, I see that it causes lots of trouble for me to debug. That is the reason I write steps like it now.

**Comment by haxxmaxx** [comment]
> > I don't mind adding the test. Maybe we need to do some shift+tab as well. But this is deviating from the the test structure in the arrow key test. Would be nice if we can write new tests in a similar way. Doesn't have to be as exactly as the arrow test. But much of the same things are going on: Do something (in this case tab) then check where we are. Should be possible to define the steps and then loop over them
> 
> I dislike the loop, I see that it causes lots of trouble for me to debug. That is the reason I write steps like it now.

Good to know, since this is a new way of writing tests that Csaba set up. Ok then it's fine writing them like this. But then there needs to be at least some comments explaining what is going on. Moving to headers, pagination controls etc

**Pull request closed without merging**